Scripts/Runtime/Entities/Tasks/TaskFlow/Nodes/JobRouteNode.cs

```diff
@@ -0,0 +1,82 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+
+namespace Anvil.Unity.DOTS.Entities.Tasks
+{
+    internal class JobRouteNode : AbstractNode
+    {
+        private readonly Dictionary<AbstractJobConfig, JobNode> m_JobsByConfig;
+        private readonly JobNodeLookup m_Lookup;
+
+        public TaskFlowRoute Route
+        {
+            get;
+        }
+
+        public JobRouteNode(JobNodeLookup lookup,
+                            TaskFlowRoute route,
+                            TaskFlowGraph taskFlowGraph,
+                            AbstractTaskSystem taskSystem,
+                            AbstractTaskDriver taskDriver) : base(taskFlowGraph, taskSystem, taskDriver)
+        {
+            m_Lookup = lookup;
+            Route = route;
+            m_JobsByConfig = new Dictionary<AbstractJobConfig, JobNode>();
+        }
+
+        protected override void DisposeSelf()
+        {
+            foreach (JobNode node in m_JobsByConfig.Values)
+            {
+                node.Dispose();
+            }
+            m_JobsByConfig.Clear();
```

Comment:
DisposeAndClear extension is your friend here.

Reply:
Beauty!